internal/pkg/api/handleEnroll.go

```diff
@@ -132,9 +132,17 @@ func (et *EnrollerT) _enroll(
 	policyID,
 	ver string) (*EnrollResponse, error) {
 
-	if req.SharedId != "" {
-		// TODO: Support pre-existing install
-		return nil, errors.New("preexisting install not yet supported")
+	var agent model.Agent
+	var enrollmentID string
+	if req.EnrollmentId != nil {
+		enrollmentID = *req.EnrollmentId
+		var err error
+		agent, err = dl.FindAgent(ctx, et.bulker, dl.QueryAgentByEnrollmentID, dl.FieldEnrollmentID, enrollmentID)
```

Comment:
This query can also fail regardless of whether an agent with this enrollment ID actually exists. Elasticsearch could be temporarily unavailable or returning 429s. Proceeding and duplicating the agent seems like the wrong choice.

As I mentioned below, I would just return the error and let the agent retry.

Reply:
It looks like the error handling was changed here and for the other two blocks below without updating any tests, which suggests that we aren't testing the failure handling this function at all.

```go
		err := invalidateAPIKey(ctx, zlog, et.bulker, agent.AccessAPIKeyID)
		if err != nil {
			zlog.Error().Err(err).
				Str("EnrollmentId", enrollmentID).
				Str("AgentId", agent.Id).
				Str("APIKeyID", agent.AccessAPIKeyID).
				Msg("Error when trying to invalidate API key of old agent with enrollment id")
			return nil, err
		}
		// delete existing agent to recreate with new api key
		err = deleteAgent(ctx, zlog, et.bulker, agent.Id)
		if err != nil {
			zlog.Error().Err(err).
				Str("EnrollmentId", enrollmentID).
				Str("AgentId", agent.Id).
				Msg("Error when trying to delete old agent with enrollment id")
			return nil, err
		}
```